### ISSUE TITLE
Static libraries

### ISSUE THREAD

**Issue by twaik** [termux/termux-packages#4028]
Hi! For now termux [removes static libraries](https://github.com/termux/termux-packages/blob/master/scripts/build/termux_step_massage.sh#L18) if package does not need it. I want to suggest a new way: keep static libraries, but pack it only into *-dev packages. It will be usable for some users that really need it on device (i.e. me :) ) or during build in another package

**Comment by Grimler91** [comment]
We could also split it into another subpackage, so having for example `libfoo-static`

**Comment by twaik** [comment]
That would be wonderful.

**Comment by twaik** [comment]
If static libraries would be subpackages are they available during ./build-package <another package>?

**Comment by xeffyr** [comment]
Static libraries will be available if `--disable-static` (which is default for now) argument is not passed to configure script.

There no matter will be they available as subpackages or not, splitting is done only after build is finished and doesn't affect existing build/install directories.

**Comment by twaik** [comment]
I think static library should be part of *-dev package. If program needs to be built with static package it also needs library headers.

**Comment by Grimler91** [comment]
> If static libraries would be subpackages are they available during ./build-package ?

Subpackages are to dependencies are automatically pulled in when building (with and without `-i`)

> I think static library should be part of *-dev package. If program needs to be built with static package it also needs library headers.

The opposite is not true however, if a program needs the library headers they don't necessarily need the static library. It would increase the -dev package's size significantly if we added the static libraries to them.

**Comment by twaik** [comment]
Ok. So libfoo-static package needs libfoo-dev package. Isn't it?

**Comment by Grimler91** [comment]
Well it's not very useful without it at least. We could make libfoo-static depend on libfoo-dev

**Comment by xeffyr** [comment]
Implemented as part of https://github.com/termux/termux-packages/commit/21b96d3b41ec130af110360759bf2117ce6b119e.
